c10/cuda/CUDAStream.cpp

```diff
@@ -18,9 +18,19 @@ namespace {
 // Internal implementation is entirely hidden
 struct CUDAStreamInternals {
   CUDAStreamInternals() = default;
+  CUDAStreamInternals(const CUDAStreamInternals&) = delete;
+  CUDAStreamInternals(CUDAStreamInternals&&) = delete;
+  CUDAStreamInternals& operator=(const CUDAStreamInternals&) = delete;
+  CUDAStreamInternals& operator=(CUDAStreamInternals&&) = delete;
 
   ~CUDAStreamInternals() {
-    if (stream) cudaStreamDestroy(stream);
+    // NB: this code is invoked only in the destruction of global variables
+    // (since we never shrink the corresponding vectors). At this point the CUDA
+    // runtime might be already destroyed and invoking cudaStreamDestroy leads
+    // to a crash. It's likely an issue in CUDA, but to be safe - let's just
+    // "forget" the destruction.
+
+    // if (stream) cudaStreamDestroy(stream);
```

Comment:
I am actually concerned that the name of the struct does not state that it's leaking streams. once the struct is destroyed the stream handle is forever lost till the end of the program lifetime. Should we, in line 18, clearly state that this is a leaky struct that is only supposed to be used inside one single place (aka holder for global streams inside this file)? Or better, name it LeakyCudaStreamInternals or something explicit.
I've seen cases when people carelessly move abstractions across files, and in the end get such tribal knowledge lost.